tests/spki_tests.rs

```diff
@@ -0,0 +1,700 @@
+                          "rsa_4096");
+// pkcs1
+test_rsa_verify_sig_file_spki!(test_rsa_verify_sig_file_spki_rsa_4096_sha256_pkcs1_2048,
+                               &spki::RSA_PKCS1_2048_8192_SHA256,
+                               "rsa_4096",
+                               "pkcs1",
+                               "sha256",
+                               "rsa_2048",
+                               "sha1");
+test_rsa_verify_sig_file_spki!(test_rsa_verify_sig_file_spki_rsa_4096_sha384_pkcs1_2048,
+                               &spki::RSA_PKCS1_2048_8192_SHA256,
+                               "rsa_4096",
+                               "pkcs1",
+                               "sha256",
+                               "rsa_2048",
+                               "sha1");
+test_rsa_verify_sig_file_spki!(test_rsa_verify_sig_file_spki_rsa_4096_sha512_pkcs1_2048,
+                               &spki::RSA_PKCS1_2048_8192_SHA256,
+                               "rsa_4096",
+                               "pkcs1",
+                               "sha256",
+                               "rsa_2048",
+                               "sha1");
+// pss
+test_rsa_verify_sig_file_spki!(test_rsa_verify_sig_file_spki_rsa_4096_sha256_pss_2048,
+                               &spki::RSA_PSS_2048_8192_SHA256_LEGACY_KEY,
+                               "rsa_4096",
+                               "pss",
+                               "sha256",
+                               "rsa_2048",
+                               "sha1");
+test_rsa_verify_sig_file_spki!(test_rsa_verify_sig_file_spki_rsa_4096_sha384_pss_2048,
+                               &spki::RSA_PSS_2048_8192_SHA256_LEGACY_KEY,
+                               "rsa_4096",
+                               "pss",
+                               "sha256",
+                               "rsa_2048",
+                               "sha1");
+test_rsa_verify_sig_file_spki!(test_rsa_verify_sig_file_spki_rsa_4096_sha512_pss_2048,
+                               &spki::RSA_PSS_2048_8192_SHA256_LEGACY_KEY,
+                               "rsa_4096",
+                               "pss",
+                               "sha256",
+                               "rsa_2048",
+                               "sha1");
+
+// 4096 bit rsa in 3072_8192 modes
+// pkcs1
+test_rsa_sign_verify_spki!(test_rsa_sign_verify_spki_rsa_4096_sha384_pkcs1_3072,
+                          &signature::RSA_PKCS1_SHA384,
+                          &spki::RSA_PKCS1_3072_8192_SHA384,
+                          "rsa_4096");
+test_rsa_verify_sig_file_spki!(test_rsa_verify_sig_file_spki_rsa_4096_sha384_pkcs1_3072,
+                               &spki::RSA_PKCS1_3072_8192_SHA384,
+                               "rsa_4096",
+                               "pkcs1",
+                               "sha384",
+                               "rsa_2048",
+                               "sha1");
+// TODO PSS for 3072-8192?
```

Comment:
Is it planned to expose PSS modes for the 3072-8192bit key RSA algorithm? If not I'll remove this todo.